src/node/blockstorage.cpp

```diff
@@ -897,7 +903,7 @@ void BlockManager::UpdateBlockInfo(const CBlock& block, unsigned int nHeight, co
     }
 
     // Update the file information with the current block.
-    const unsigned int added_size = ::GetSerializeSize(TX_WITH_WITNESS(block));
+    const auto added_size{static_cast<uint32_t>(GetSerializeSize(TX_WITH_WITNESS(block)))};
```

Comment:
It feels off to cast of client side bacause we know it would be serialized incorrectly if we actually used the 64 bit version. It seems like a code smell, it seems to me we should go the other way: return the actual size and only cast up before the problematic multiplications.

Reply:
Yeah, I can go that way, but I think using 64-bit by default and explicitly casting after the range has been properly checked seems safer, see also https://github.com/bitcoin/bitcoin/pull/33724#discussion_r2476885390

Reply:
https://github.com/bitcoin/bitcoin/pull/33724#discussion_r2476885390 explains it well, I can get behind both as long as we're consistent

Reply:
Dropped the `blockstorage.cpp` changes